src/main/java/org/commcare/formplayer/sandbox/ArchivableFile.java

```diff
@@ -71,6 +72,9 @@ private static void decompressGzipFile(File gzipFile, File newFile) throws IOExc
             int len;
             while ((len = gis.read(buffer)) != -1) {
                 fos.write(buffer, 0, len);
+                if (Thread.interrupted()) {
+                    throw new RequestAbandonedException();
```

Comment:
InterruptedRuntimeException would be a better choice here. This exception specifically is intended to signal cleanup during XPath Query evaluation.,

Reply:
I had originally used InterruptedRuntimeException but it requires the InterruptException to [be passed in](https://github.com/dimagi/formplayer/blob/d1e8cba7187ad93248bd4e2440d920b6f0153ffa/src/main/java/org/commcare/formplayer/sandbox/ArchivableFile.java#L111). We can always create a constructor that doesn't require that exception, but I wasn't sure if semantically an InterruptedRuntimeException is tied to an InterruptedException.

Do you also want me to change the 2 SqlStorage references (e.g. https://github.com/dimagi/formplayer/blob/d1e8cba7187ad93248bd4e2440d920b6f0153ffa/src/main/java/org/commcare/formplayer/sandbox/SqlStorage.java#L479) too?

Reply:
Hm, the SqlStorage references throw `RequestAbandonedException` types because they are frequently used in XPath references. Now that we are discussing things, it occurs to me that the `ArchivableFile` errors here are actually part of a `SqlStorage` callstack in the first place, so my initial read was wrong. I now think it makes sense for these to be RequestAbandoned throws. The alternative would probably be throwing an InterruptedException up the stack for the SqlStorage to catch and rethrow, but I don't think that would add much value. 

Thanks for thinking through it.